talon/utils.py

```diff
@@ -46,8 +48,8 @@ def to_unicode(str_or_unicode, precise=False):
     If `precise` flag is True, tries to guess the correct encoding first.
     """
     encoding = quick_detect_encoding(str_or_unicode) if precise else 'utf-8'
```

Comment:
Can you please move it inside the ``if`` statement below: with the ``assert`` added to ``quick_detect_encoding`` and ``detect_encoding`` this function raises the ``AsserionException`` if the passed string is already in unicode.